pages/curriculum.tsx

```diff
@@ -4,76 +4,75 @@ import LessonCard from '../components/LessonCard'
 import ProgressCard from '../components/ProgressCard'
 import AnnouncementCard from '../components/AnnouncementCard'
 import AdditionalResources from '../components/AdditionalResources'
-import { Lesson, LessonStatus } from '../@types/lesson'
-import { AppData } from '../@types/app'
-import GET_APP from '../graphql/queries/getApp'
-import withQueryLoader, { QueryDataProps } from '../containers/withQueryLoader'
 import AlertsDisplay from '../components/AlertsDisplay'
+import LoadingSpinner from '../components/LoadingSpinner'
+import { withGetApp, GetAppProps } from '../graphql/'
 import _ from 'lodash'
 
-type LessonStatusMap = {
-  [id: string]: LessonStatus
-}
+export const Curriculum: React.FC<GetAppProps> = ({ data }) => {
+  let { loading, error, alerts, lessons, session } = data
+
+  if (loading) return <LoadingSpinner />
+  if (error) return <h1>Error</h1>
+  if (!session || !lessons || !alerts) return <h1>Bad Data</h1>
 
-export const Curriculum: React.FC<QueryDataProps<AppData>> = ({
-  queryData
-}) => {
   const announcements = [
     'To make space for other students on our servers, your account will be deleted after 30 days of inactivity.',
     'Take each lesson challenge seriously and do them over and over again until you can solve them. With the exception End to End, all challenges are questions and exercises taken from real interviews.',
     'This lesson will not only prepare you for interviews, but it will also help teach you the skills that you need to become an effective engineer.',
     'After completing Foundations of JavaScript, Variables & Functions, Array, Objects, End to End, HTML/CSS/JavaScript, React/GraphQL/SocketIO, you will be technically ready to contribute to our codebase.'
   ]
 
-  const { lessons, session, alerts } = queryData
-  const lessonStatus: LessonStatus[] = _.get(session, 'lessonStatus', [])
-  const lessonStatusMap: LessonStatusMap = lessonStatus.reduce(
-    (map: LessonStatusMap, lessonStatus: LessonStatus) => {
-      map[lessonStatus.lessonId] = lessonStatus
-      return map
-    },
-    {}
-  )
-
-  const lessonsWithStatus: Lesson[] = lessons.map((lesson: Lesson) => {
-    lesson.lessonStatus = lessonStatusMap[lesson.id] || {
-      isEnrolled: null,
-      isTeaching: null,
-      lessonId: lesson.id
-    }
-    return lesson
-  })
+  const { lessonStatus } = session
+  const lessonStatusMap: { [id: string]: typeof lessonStatus[0] } = {}
+  for (const status of lessonStatus) {
+    const lessonId = _.get(status, 'lessonId', '-1') as string
+    lessonStatusMap[lessonId] = status
+  }
 
-  const lessonInProgressIdx =
-    lessonsWithStatus.findIndex(lesson => !lesson.lessonStatus.isPassed) || 0
+  const lessonInProgressIdx = _.cond([
+    [_.isEqualWith.bind(null, -1), _.constant(0)],
```

Comment:
I'm learning some new `lodash` methods today :)

In this case, when we have no `customizer`, what advantage does `isEqualWith` provide over `isEqual`?

Reply:
Very good point! haha I'm was learning these as I went as well lol